components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -3119,8 +3119,13 @@ public static String getIdTokenIssuer(String tenantDomain) throws IdentityOAuth2
         FederatedAuthenticatorConfig oidcAuthenticatorConfig =
                 IdentityApplicationManagementUtil.getFederatedAuthenticator(fedAuthnConfigs,
                         IdentityApplicationConstants.Authenticator.OIDC.NAME);
-        return IdentityApplicationManagementUtil.getProperty(oidcAuthenticatorConfig.getProperties(),
+        String idpEntityId = IdentityApplicationManagementUtil.getProperty(oidcAuthenticatorConfig.getProperties(),
                 IDP_ENTITY_ID).getValue();
+        if (idpEntityId.equals(IdentityUtil.getProperty("OAuth.OpenIDConnect.IDTokenIssuerID"))) {
```

Comment:
This will lead to NPE if idpEntityId evaluates to null. Can we be sure that it can never be null?
Also can IdentityUtil.getProperty("OAuth.OpenIDConnect.IDTokenIssuerID") evaluate to null in any scenario?